widget/entry.go

```diff
@@ -898,6 +898,38 @@ func (e *Entry) registerShortcut() {
 	e.shortcut.AddShortcut(&fyne.ShortcutSelectAll{}, func(se fyne.Shortcut) {
 		e.selectAll()
 	})
+
+	moveWord := func(s fyne.Shortcut) {
+		row := e.textProvider().row(e.CursorRow)
+		start, end := getTextWhitespaceRegion(row, e.CursorColumn, true)
+		if start == -1 || end == -1 {
+			return
+		}
+
+		e.setFieldsAndRefresh(func() {
+			if s.(*desktop.CustomShortcut).KeyName == fyne.KeyLeft {
+				if e.CursorColumn == 0 {
+					if e.CursorRow > 0 {
+						e.CursorRow--
+						e.CursorColumn = len(e.textProvider().row(e.CursorRow))
+					}
+				} else {
+					e.CursorColumn = start
+				}
+			} else {
+				if e.CursorColumn == len(e.textProvider().row(e.CursorRow)) {
+					if e.CursorRow < e.textProvider().rows()-1 {
```

Comment:
Same as comment above.

Reply:
As before - the else applies to the outer `if`, not to both.

Reply:
It should be equivalent to just do `e.CursorColumn == len(e.textProvider().row(e.CursorRow)) && e.CursorRow < e.textProvider().rows()-1`.

Reply:
I don't see why this would be wrong. Avoids the triple if statements and instead keeps it down to two.
```go
if e.CursorColumn == len(e.textProvider().row(e.CursorRow)) && e.CursorRow < e.textProvider().rows()-1 {

} else {

}
```

Reply:
As above it will change the action for one of the cases. Are you sure that is a desirable change?

Reply:
You are right. Sorry for the fuss